pkg/crc/config/config.go

```diff
@@ -95,6 +96,36 @@ func (c *Config) Set(key string, value interface{}) (string, error) {
 		return "", fmt.Errorf(invalidType, value, key)
 	}
 
+	// Preset is mapped with `memory`, `cpus` and `bundle` and
+	// we want to make sure if cpu or memory is less for a preset
+	// then default is set automatic.
+	if setting.Name == Preset {
+		preset := preset.ParsePreset(value.(string))
+		mem := c.Get(Memory)
+		if err := validation.ValidateMemory(mem.AsInt(), preset); err != nil {
+			if _, err := c.Unset(Memory); err != nil {
+				return "", err
+			}
+		}
+		cpu := c.Get(CPUs)
+		if err := validation.ValidateCPUs(cpu.AsInt(), preset); err != nil {
+			if _, err := c.Unset(CPUs); err != nil {
+				return "", err
+			}
+		}
+	}
+
+	// Make sure if user try to set same value which
+	// is default then just unset the value which
+	// anyway make it default and don't update it
+	// ~/.crc/crc.json (viper config) file.
+	if setting.defaultValue == castValue {
+		if _, err := c.Unset(key); err != nil {
+			return "", err
+		}
+		return "", nil
```

Comment:
i think it'd be better to include a message letting the user know the config is restored to its default, since we won't  see the config option in `config view` ?

Reply:
@anjannath Currently also we are not seeing the default configs as part of `config view`.

Reply:
right, do you prefer to keep it that way, or show a message saying `Configuration property '%s' restored to default value '%s'` (since you are in the file ;) )

Reply:
@anjannath better to keep it that way.

Reply:
I also think we need to inform the user that we changed their configuration settings. I'll add that.